server/ServerNetworking.cpp

```diff
@@ -188,7 +188,14 @@ PlayerConnection::~PlayerConnection() {
                                  << " for player id " << m_ID;
         }
     }
-    m_socket.close();
+    boost::optional<boost::asio::ip::tcp::socket> socket;
+    int id = m_ID;
+    boost::swap(m_socket, socket);
+    std::async(std::launch::async, [socket{std::move(socket)}, id] () mutable {
```

Comment:
According to [this](https://en.cppreference.com/w/cpp/thread/async), "If the std::future obtained from std::async is not moved from or bound to a reference, the destructor of the std::future will block at the end of the full expression until the asynchronous operation completes..."

So do you need to do something with the return value here to avoid blocking until after the `close` call?

Reply:
Looks like I need other way to call `close` in the separate thread.

Reply:
Can you just launch a thread? `std::async` is set up to return something in a `std::future` but you don't need to wait for a return value...

Reply:
I found out boost get thread pool starting 1.66 version, but I use `std::thread` for now.

Reply:
I think a thread pool would be inappropriate here... You don't have many small tasks, but rather just one, which you want to run asynchronously, so the overhead to launch a thread for that isn't much of a problem.

Just launching a thread (and detaching it) should be fine...